.github/workflows/auto-merge.yml

```diff
@@ -42,13 +42,13 @@ jobs:
           git config user.email "70000568+nvauto@users.noreply.github.com "
           git fetch origin ${HEAD} ${BASE}
           git checkout -b ${INTERMEDIATE_HEAD} origin/${HEAD}
-          OUT=$(git --no-pager diff --name-only origin/${BASE} | grep "${FILE_USE_BASE}" || true)
-          [[ ! -z "${OUT}" ]] && git checkout origin/${BASE} -- ${FILE_USE_BASE} && \
-            git commit -s -am "Auto-merge use submodule in BASE ref"
+          git checkout origin/${BASE} -- ${FILE_USE_BASE}
```

Comment:
Please add somewhere comments or a doc explaining at the high level what is happening here.

Reply:
> Please add somewhere comments or a doc explaining at the high level what is happening here. Among others, it should reduce the chance of the previous confusion if we want an empty git status or not.

Added comments to explain what we want to do here

https://github.com/NVIDIA/spark-rapids-jni/pull/1928/commits/22314bc90be2315b58565058ef0b862b0a369e58